### ISSUE TITLE
I need to filter some words, to get rid of these keywords,but...Who can help me

### ISSUE THREAD

**Issue by lwpaxcdcg** [SpiderLabs/owasp-modsecurity-crs#686]
I met a strange question, modsecurity + apache2.2, with the following rules, access to the page. If this is the index HTML or index. TXT, rules can take effect.
If the index instead. PHP or index. 456 and so on, the rule will not trigger.

SecRule RESPONSE_BODY "@rx test|大纪元|.....\
        "id:350000,\
        phase:4,\
        t:none,\
        log,\
        capture,\
        redirect:'http://xxx.com/keyword_filter.php?keys=%{TX.0}',\
        msg:'keyword filter %{TX.0}'"

		
modsecurity.confSet the following:

SecResponseBodyAccess On
SecResponseBodyMimeType text/plain text/html text/xml
SecResponseBodyLimit 524288
SecResponseBodyLimitAction ProcessPartial
		
		
		
debugLog as follows：

Access to 123.html，Target value "xmp08 test page"：
		
		[17/Feb/2017:11:08:43 +0800] [test.xmp08.xxx.com/sid#2a77c90][rid#3b372b0][/123.html][4] Recipe: Invoking rule 2a2ca18; [file "/usr/local/modsecurity/modsecurity-crs/rules/modsecurity_crs_15_customrules.conf"] [line "9"] [id "350000"].
[17/Feb/2017:11:08:43 +0800] [test.xmp08.xxx.com/sid#2a77c90][rid#3b372b0][/123.html][5] Rule 2a2ca18: SecRule "RESPONSE_BODY" "@rx test|\xe5\xa4\xa7\xe7\xba\xaa\xe5\x85\x83|\xe7\xa0\xb4\xe7\xbd\x91\xe8\xbd\xaf\xe4\xbb\xb6|\xe6\xb2\x88\xe9\x98\xb3\xe8\x8b\x8f\xe5\xae\xb6\xe5\xb1\xaf|\xe5\x99\xa8\xe5\xae\x98\xe4\xb9\xb0\xe5\x8d\x96|\xe4\xb9\x9d\xe8\xaf\x84\xe5\x85\xb1\xe4\xba\xa7\xe5\x85\x9a|\xe4\xb8\x89\xe5\x94\x91\xe4\xbb\x91|\xe6\x81\x90\xe6\x80\x96\xe6\x9a\xb4\xe8\xa1\x8c|\xe6\x97\xa0\xe7\x95\x8c\xe6\xb5\x8f\xe8\xa7\x88\xe5\x99\xa8|\xe6\xb3\x95\xe8\xbd\xae\xe5\x8a\x9f|\xe9\xab\x98\xe6\x99\xba\xe6\x99\x9f|\xe7\xaa\x81\xe7\xa0\xb4\xe7\xbd\x91\xe7\xbb\x9c\xe5\xb0\x81\xe9\x94\x81|\xe7\xa0\xb4\xe5\xb0\x81\xe9\x94\x81|\xe6\xb3\x95\xe8\xbc\xaa\xe5\x8a\x9f|\xe5\xa4\xa7\xe6\xb3\x95\xe5\xbc\x9f\xe5\xad\x90|\xe6\x98\x8e\xe6\x85\xa7\xe7\xbd\x91|\xe6\x98\x8e\xe6\x85\xa7\xe7\xb6\xb2|\xe4\xba\x8c\xe5\xa5\xb6\xe5\xa4\xa7\xe5\xa5\x96\xe8\xb5\x9b|\xe8\xa7\x82\xe9\x9f\xb3\xe6\xb3\x95\xe9\x97\xa8|\xe9\x9b\xb6\xe5\x85\xab\xe5\xae\xaa\xe7\xab\xa0|08\xe5\xae\xaa\xe7\xab\xa0|\xe6\xaf\x9b\xe7\x88\xb7\xe7\x88\xb7\xe5\
[17/Feb/2017:11:08:43 +0800] [test.xmp08.xxx.com/sid#2a77c90][rid#3b372b0][/123.html][4] Transformation completed in 0 usec.
[17/Feb/2017:11:08:43 +0800] [test.xmp08.xxx.com/sid#2a77c90][rid#3b372b0][/123.html][4] Executing operator "rx" with param "test|\xe5\xa4\xa7\xe7\xba\xaa\xe5\x85\x83|\xe7\xa0\xb4\xe7\xbd\x91\xe8\xbd\xaf\xe4\xbb\xb6|\xe6\xb2\x88\xe9\x98\xb3\xe8\x8b\x8f\xe5\xae\xb6\xe5\xb1\xaf|\xe5\x99\xa8\xe5\xae\x98\xe4\xb9\xb0\xe5\x8d\x96|\xe4\xb9\x9d\xe8\xaf\x84\xe5\x85\xb1\xe4\xba\xa7\xe5\x85\x9a|\xe4\xb8\x89\xe5\x94\x91\xe4\xbb\x91|\xe6\x81\x90\xe6\x80\x96\xe6\x9a\xb4\xe8\xa1\x8c|\xe6\x97\xa0\xe7\x95\x8c\xe6\xb5\x8f\xe8\xa7\x88\xe5\x99\xa8|\xe6\xb3\x95\xe8\xbd\xae\xe5\x8a\x9f|\xe9\xab\x98\xe6\x99\xba\xe6\x99\x9f|\xe7\xaa\x81\xe7\xa0\xb4\xe7\xbd\x91\xe7\xbb\x9c\xe5\xb0\x81\xe9\x94\x81|\xe7\xa0\xb4\xe5\xb0\x81\xe9\x94\x81|\xe6\xb3\x95\xe8\xbc\xaa\xe5\x8a\x9f|\xe5\xa4\xa7\xe6\xb3\x95\xe5\xbc\x9f\xe5\xad\x90|\xe6\x98\x8e\xe6\x85\xa7\xe7\xbd\x91|\xe6\x98\x8e\xe6\x85\xa7\xe7\xb6\xb2|\xe4\xba\x8c\xe5\xa5\xb6\xe5\xa4\xa7\xe5\xa5\x96\xe8\xb5\x9b|\xe8\xa7\x82\xe9\x9f\xb3\xe6\xb3\x95\xe9\x97\xa8|\xe9\x9b\xb6\xe5\x85\xab\xe5\xae\xaa\xe7\xab\xa0|08\xe5\xae\xaa\xe7\xab\xa0|\xe6\xaf\x9b\xe7\x88\xb7\xe7\x88\xb7\xe5\xa4\x8d
[17/Feb/2017:11:08:43 +0800] [test.xmp08.xxx.com/sid#2a77c90][rid#3b372b0][/123.html][9] Target value: "xmp08 test page\n"
[17/Feb/2017:11:08:43 +0800] [test.xmp08.xxx.com/sid#2a77c90][rid#3b372b0][/123.html][9] Added regex subexpression to TX.0: test
[17/Feb/2017:11:08:43 +0800] [test.xmp08.xxx.com/sid#2a77c90][rid#3b372b0][/123.html][4] Operator completed in 66 usec.
[17/Feb/2017:11:08:43 +0800] [test.xmp08.xxx.com/sid#2a77c90][rid#3b372b0][/123.html][9] Resolved macro %{TX.0} to: test
[17/Feb/2017:11:08:43 +0800] [test.xmp08.xxx.com/sid#2a77c90][rid#3b372b0][/123.html][9] Resolved macro %{TX.0} to: test
[17/Feb/2017:11:08:43 +0800] [test.xmp08.xxx.com/sid#2a77c90][rid#3b372b0][/123.html][4] Rule returned 1.
[17/Feb/2017:11:08:43 +0800] [test.xmp08.xxx.com/sid#2a77c90][rid#3b372b0][/123.html][9] Match, intercepted -> returning.

123.html rename123.php,Target value: "" :

[17/Feb/2017:11:07:29 +0800] [test.xmp08.xxx.com/sid#2a77c90][rid#3b372b0][/123.456][4] Recipe: Invoking rule 2a2ca18; [file "/usr/local/modsecurity/modsecurity-crs/rules/modsecurity_crs_15_customrules.conf"] [line "9"] [id "350000"].
[17/Feb/2017:11:07:29 +0800] [test.xmp08.xxx.com/sid#2a77c90][rid#3b372b0][/123.456][5] Rule 2a2ca18: SecRule "RESPONSE_BODY" "@rx test|\xe5\xa4\xa7\xe7\xba\xaa\xe5\x85\x83|\xe7\xa0\xb4\xe7\xbd\x91\xe8\xbd\xaf\xe4\xbb\xb6|\xe6\xb2\x88\xe9\x98\xb3\xe8\x8b\x8f\xe5\xae\xb6\xe5\xb1\xaf|\xe5\x99\xa8\xe5\xae\x98\xe4\xb9\xb0\xe5\x8d\x96|\xe4\xb9\x9d\xe8\xaf\x84\xe5\x85\xb1\xe4\xba\xa7\xe5\x85\x9a|\xe4\xb8\x89\xe5\x94\x91\xe4\xbb\x91|\xe6\x81\x90\xe6\x80\x96\xe6\x9a\xb4\xe8\xa1\x8c|\xe6\x97\xa0\xe7\x95\x8c\xe6\xb5\x8f\xe8\xa7\x88\xe5\x99\xa8|\xe6\xb3\x95\xe8\xbd\xae\xe5\x8a\x9f|\xe9\xab\x98\xe6\x99\xba\xe6\x99\x9f|\xe7\xaa\x81\xe7\xa0\xb4\xe7\xbd\x91\xe7\xbb\x9c\xe5\xb0\x81\xe9\x94\x81|\xe7\xa0\xb4\xe5\xb0\x81\xe9\x94\x81|\xe6\xb3\x95\xe8\xbc\xaa\xe5\x8a\x9f|\xe5\xa4\xa7\xe6\xb3\x95\xe5\xbc\x9f\xe5\xad\x90|\xe6\x98\x8e\xe6\x85\xa7\xe7\xbd\x91|\xe6\x98\x8e\xe6\x85\xa7\xe7\xb6\xb2|\xe4\xba\x8c\xe5\xa5\xb6\xe5\xa4\xa7\xe5\xa5\x96\xe8\xb5\x9b|\xe8\xa7\x82\xe9\x9f\xb3\xe6\xb3\x95\xe9\x97\xa8|\xe9\x9b\xb6\xe5\x85\xab\xe5\xae\xaa\xe7\xab\xa0|08\xe5\xae\xaa\xe7\xab\xa0|\xe6\xaf\x9b\xe7\x88\xb7\xe7\x88\xb7\xe5\
[17/Feb/2017:11:07:29 +0800] [test.xmp08.xxx.com/sid#2a77c90][rid#3b372b0][/123.456][4] Transformation completed in 1 usec.
[17/Feb/2017:11:07:29 +0800] [test.xmp08.xxx.com/sid#2a77c90][rid#3b372b0][/123.456][4] Executing operator "rx" with param "test|\xe5\xa4\xa7\xe7\xba\xaa\xe5\x85\x83|\xe7\xa0\xb4\xe7\xbd\x91\xe8\xbd\xaf\xe4\xbb\xb6|\xe6\xb2\x88\xe9\x98\xb3\xe8\x8b\x8f\xe5\xae\xb6\xe5\xb1\xaf|\xe5\x99\xa8\xe5\xae\x98\xe4\xb9\xb0\xe5\x8d\x96|\xe4\xb9\x9d\xe8\xaf\x84\xe5\x85\xb1\xe4\xba\xa7\xe5\x85\x9a|\xe4\xb8\x89\xe5\x94\x91\xe4\xbb\x91|\xe6\x81\x90\xe6\x80\x96\xe6\x9a\xb4\xe8\xa1\x8c|\xe6\x97\xa0\xe7\x95\x8c\xe6\xb5\x8f\xe8\xa7\x88\xe5\x99\xa8|\xe6\xb3\x95\xe8\xbd\xae\xe5\x8a\x9f|\xe9\xab\x98\xe6\x99\xba\xe6\x99\x9f|\xe7\xaa\x81\xe7\xa0\xb4\xe7\xbd\x91\xe7\xbb\x9c\xe5\xb0\x81\xe9\x94\x81|\xe7\xa0\xb4\xe5\xb0\x81\xe9\x94\x81|\xe6\xb3\x95\xe8\xbc\xaa\xe5\x8a\x9f|\xe5\xa4\xa7\xe6\xb3\x95\xe5\xbc\x9f\xe5\xad\x90|\xe6\x98\x8e\xe6\x85\xa7\xe7\xbd\x91|\xe6\x98\x8e\xe6\x85\xa7\xe7\xb6\xb2|\xe4\xba\x8c\xe5\xa5\xb6\xe5\xa4\xa7\xe5\xa5\x96\xe8\xb5\x9b|\xe8\xa7\x82\xe9\x9f\xb3\xe6\xb3\x95\xe9\x97\xa8|\xe9\x9b\xb6\xe5\x85\xab\xe5\xae\xaa\xe7\xab\xa0|08\xe5\xae\xaa\xe7\xab\xa0|\xe6\xaf\x9b\xe7\x88\xb7\xe7\x88\xb7\xe5\xa4\x8d
[17/Feb/2017:11:07:29 +0800] [test.xmp08.xxx.com/sid#2a77c90][rid#3b372b0][/123.456][9] Target value: ""
[17/Feb/2017:11:07:29 +0800] [test.xmp08.xxx.com/sid#2a77c90][rid#3b372b0][/123.456][4] Operator completed in 7 usec.
[17/Feb/2017:11:07:29 +0800] [test.xmp08.xxx.com/sid#2a77c90][rid#3b372b0][/123.456][4] Rule returned 0.
[17/Feb/2017:11:07:29 +0800] [test.xmp08.xxx.com/sid#2a77c90][rid#3b372b0][/123.456][9] No match, not chained -> mode NEXT_RULE.
[17/Feb/2017:11:07:29 +0800] [test.xmp08.xxx.com/sid#2a77c90][rid#3b372b0][/123.456][4] Output filter: Output forwarding complete.
[17/Feb/2017:11:07:29 +0800] [test.xmp08.xxx.com/sid#2a77c90][rid#3b372b0][/123.456][9] Output filter: Sending input brigade directly.
[17/Feb/2017:11:07:29 +0800] [test.xmp08.xxx.com/sid#2a77c90][rid#3b372b0][/123.456][4] Initialising logging.
[17/Feb/2017:11:07:29 +0800] [test.xmp08.xxx.com/sid#2a77c90][rid#3b372b0][/123.456][4] Starting phase LOGGING.
[17/Feb/2017:11:07:29 +0800] [test.xmp08.xxx.com/sid#2a77c90][rid#3b372b0][/123.456][9] This phase consists of 50 rule(s).
[17/Feb/2017:11:07:29 +0800] [test.xmp08.xxx.com/sid#2a77c90][rid#3b372b0][/123.456][4] Recipe: Invoking rule 34274d0; [file "/usr/local/modsecurity/modsecurity-crs/rules/REQUEST-912-DOS-PROTECTION.conf"] [line "84"] [id "912110"].
[17/Feb/2017:11:07:29 +0800] [test.xmp08.xxx.com/sid#2a77c90][rid#3b372b0][/123.456][5] Rule 34274d0: SecRule "&TX:dos_burst_time_slice" "@eq 0" "phase:5,auditlog,id:912110,t:none,nolog,chain,skipAfter:END_DOS_PROTECTION_CHECKS"

**Comment by ghost** [comment]
Why was I given a notification for this?

**Comment by lwpaxcdcg** [comment]
I don't know why I can't get a proper notice, sometimes test.xmp08.xxx.com/favicon.ico this file received notice, but it's not regular, and not I hope to get results.What are the likely cause I like this weird question?Apache configuration might be the problem cause?
How can see me in the debug log SecRequestBody cache content?The debug log level is set to 9

**Comment by dune73** [comment]
@eq: The ModSecurity rule language has an operator `@eq`. It's rarely used on github, but notices can happen. The poor user with the handle rx gets notices all the time as rx is our the ModSecurity standard operator.

**Comment by dune73** [comment]
@lwpaxcdcg this is not a Core Rule Set question, as it is a question about a rule you wrote yourself. It might be better to address it to the ModSecurity user mailinglist.

Your English makes the exact question a bit hard to grasp. I am not totally sure what you want to tell me.

I'll try to give it a shot nevertheless. Your first debug log entry describes a response where ModSec gets a  RESPONSE_BODY, the 2nd time around it does not. You can display the value in the error log if you want to :  `SecAction ..... msg:%{RESPONSE_BODY}"`. The fact that the response body is empty is likely due to the response mime type setting, but I am not really sure.

Closing this now. Good luck.

**Comment by lwpaxcdcg** [comment]
dear dune73:
I've found the reason why php file don't hit customized rules.
I've add a MIME in httpd.conf:
    AddType application/x-httpd-php .php .php4 .php3 .phtml .phtm .asmx .dll .aspx .asp
    Action  application/x-httpd-php /cgi-bin/.parsephp
	
1.On one hand is I didn't add application/x-httpd-php to SecResponseBodyMimeType (already add now).
2.On the other hand is about Action in httpd.conf, rules can only take effect when I commented Action.

Buuuuuut, If I comment Action, then PHP won't work anymore, I was wondering if there is any way to solve this problem?
